std/demangle.d

```diff
@@ -18,72 +18,51 @@
  */
 module std.demangle;
 
-/+
-private class MangleException : Exception
+/**
+Demangle D mangled names.
+
+Params:
+    name = the mangled name
+Returns:
+    A `string`. If it is not a D mangled name, it returns its argument name.
+ */
+string demangle(string name) @safe pure nothrow
 {
-    this()
-    {
-        super("MangleException");
-    }
+    import core.demangle : demangle;
+    import std.exception : assumeUnique;
+    auto ret = demangle(name);
+    return () @trusted { return ret.assumeUnique; } ();
 }
-+/
-
-/*****************************
- * Demangle D mangled names.
- *
- * If it is not a D mangled name, it returns its argument name.
- * Example:
- *        This program reads standard in and writes it to standard out,
- *        pretty-printing any found D mangled names.
--------------------
-import core.stdc.stdio : stdin;
-import std.stdio;
-import std.ascii;
-import std.demangle;
 
-void test(int x, float y) { }
+///
+@safe pure unittest
+{
+    // int b in module a
+    assert(demangle("_D1a1bi") == "int a.b");
+    // char array foo in module test
+    assert(demangle("_D4test3fooAa") == "char[] test.foo");
+}
 
-int main()
+/**
+This program reads standard in and writes it to standard out,
+pretty-printing any found D mangled names.
+ */
+@system unittest
 {
-    string buffer;
-    bool inword;
-    int c;
+    import std.ascii;
+    import std.algorithm;
+    import std.conv;
+    import std.demangle;
+    import std.functional;
+    import std.stdio;
```

Comment:
What's the reason for using individual lines here? The imports are now longer than the actual example :O

Reply:
I'm not against it in principal, but the vast majority of code in Phobos uses one import per-line style.

Reply:
I don't have a strong opinion either, but the vast majority of Phobos code uses selective imports too.
I just checked dlang.org and we use one-line imports for most examples there (except for the ones with selective imports).
That being said, I don't feel strongly about this.